src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+using Parameters
+using SpecialFunctions: erfc
+using .GLMakie
+
+export StrengthEnvelope, ConstantTemp, LinearTemp, HalfspaceCoolingTemp, GP_Compute_ThermalStructure, LithPres
+
+abstract type AbstractThermalStructure end  
+
+"""
+    ConstantTemp(T=1000)
+    
+Sets a constant temperature inside the box
+Parameters
+===
+- T : the value
+"""
+@with_kw_noshow mutable struct ConstantTemp <: AbstractThermalStructure
+    T = 1000
+end
+
+function GP_Compute_ThermalStructure(Z, s::ConstantTemp)
+    Temp  = zeros(Float64, length(Z))
+    Temp .= s.T
+    return Temp
+end
+
+
+"""
+    LinearTemp(Ttop=0, Tbot=1000)
+    
+Set a linear temperature structure from top to bottom
+Parameters
+===
+- Ttop : the value @ the top
+- Tbot : the value @ the bottom
+"""
+@with_kw_noshow mutable struct LinearTemp <: AbstractThermalStructure
+    Ttop = 0
+    Tbot = 1350
+end
+
+function GP_Compute_ThermalStructure(Z, s::LinearTemp)
+    @unpack Ttop, Tbot  = s
+
+    Temp  = zeros(Float64, length(Z))
+
+    dz    = Z[end]-Z[1];
+    dT    = Tbot - Ttop
+
+    Temp .= abs.(Z./dz).*dT .+ Ttop
```

Comment:
You don't really need to initialize `Temp` above,  just do `Temp = @.  abs(Z/dz)*dT + Ttop`